src/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -67,6 +67,9 @@
     // If true, all SAMRecords are fully decoded as they are read.
     private boolean eagerDecode;
 
+    // If true, the BAMFileReader will use asynchronous IO.
+    private final boolean useAsynchronousIO;
```

Comment:
Added comment.

The field is final so it can't default to anything - by design, it's the factory's responsibility to set the reader in a correct way and the reader should not reach out to the defaults.